### PR TITLE
Add features for the front end app in the full-stack capstone project - Tesla Car Test Drive Booking

### PR DESCRIPTION
**Pull request by ivangonzalez224**
## Final PR - Front end app for Full Stack Capstone project - Tesla Car Test Drive Booking

**Description:**

This pull request implements the following features for the Tesla car test drive booking front end app:

- Installed react-redux, react-router, react-icons, axios and the necessary dependencies.
- Added the store and the reducers.
- Added dynamic routing for the whole project.
- Added the components for each page.
- Set up the structure of navigation panel, the sign in, sign out, the main, details, reserve form, my reservations and create and delete car pages.
- Added the functionality to sign in and sign out in a proper way by fetching the authentication APIs.
- Fetched the cars, reservations and cities data from the API.
- Displayed the list of cars on the main page.
- Added logic to filter reservations by user ID and is_remove cars column.
- Rendered the filtered data in my reservations page.
- Added styles to the navigation panel, the main, details, reserve form, my reservations and create and delete car pages.

**Technical Setup:**

Gitflow workflow implemented.
React & Redux for frontend UI.

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,24 +1,33 @@
...
+import AddCarsForm from './components/cars/AddCarsForm';
+import Session from './components/session/Session';
+import MainPage from './components/cars/MainPage';
+import DeleteCar from './components/cars/DeleteCar';
+import Reservations from './components/reservations/Reservations';
+import CarDetails from './components/details/CarDetails';
+import ReserveNav from './components/reserve/ReserveNav';
+import ReserveDetails from './components/reserve/ReserveDetails';
 
 function App() {
```
Comment (Meltrust) [inline src/App.js:15]: - [ ] This is the test that comes with React:

image

Kindly either implement it or remove it (the file) if not using it. 👍

Comment (Meltrust) [inline src/App.js:15]: - [ ]  Please, kindly refactor this main "App" functional component to ES6 arrow function, so your code complies with the latest standards.

Refactoring is easy:

```javascript
const App = () => {
  // ...
}
```

## Explanation

The `function` keyword has been soft deprecated in almost all the software industry in favor of ES6 arrow functions. This means, almost nobody uses it anymore, so much that it's considered a non-best practice.

That way, your code will meet the latest standards. 💪

File: src/App.js [hunk 2]
```diff
@@ -1,24 +1,33 @@
-import logo from './logo.svg';
-import './App.css';
+import React from 'react';
```
Comment (Meltrust) [inline src/App.js:1]: - [ ] When adding a new car, the image is inconsistent with the others:

image

Kindly, add some boundaries to your CSS styling, so any user added image is the same size as the others. 👍

**Review by Meltrust — CHANGES_REQUESTED** [review]
Hi,

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation, but you are almost there!






### To highlight:

-  Nice validations ✔️ 



#### You are really close to finishing the Microverse program!! Keep it up! 👍👍👍

You can do it

### After implementing the requested changes, please submit another review request. ♻️
_Check the comments under the review._







Cheers and Happy coding!👏👏👏

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the previous reviews unless it is requested otherwise._**

------

**Comment by ivangonzalez224** [comment]
Hi @Meltrust thank you for the review
Changes done: refactored the app component to meet the latest standards, removed the test file and fixed the styles in Main page.

**Review by Meltrust — APPROVED** [review]
# **Wow, you did it 🎉**
Brilliant
## Thank you for the changes implemented 💪 🥇 ㊗️ 
#### _Unless you want to add more features, go ahead to your final presentation ⏩ ⏩ ⏩_ 
## You are about to finish the Microverse program.  You have come a long way!!!




### Good luck in the software industry!!  I'll see you there. ✨
### **Congratulations!!!!!!**  🎉
applause

### To highlight


- Superb job ✔️  



Cheers and Happy coding!👏👏👏


------

**Pull request merged**